release.sh

```diff
@@ -313,64 +314,105 @@ function build_from_source() {
 }
 
 function get_images_in_yamls() {
-  rm -rf imagerefs.txt
+  rm -rf "$IMAGES_REFS_FILE"
   echo "Assembling a list of image refences to sign"
-  for file in $@; do
+  for file in "$@"; do
     [[ "${file##*.}" != "yaml" ]] && continue
     echo "Inspecting ${file}"
-    for image in $(grep -oh "\S*${KO_DOCKER_REPO}\S*" "${file}"); do
-      echo $image >> imagerefs.txt
-    done
+    while read -r image; do
+      echo "$image" >> "$IMAGES_REFS_FILE"
+    done < <(grep -oh "\S*${KO_DOCKER_REPO}\S*" "${file}")
+  done
+  if [[ -f "$IMAGES_REFS_FILE" ]]; then
+    sort -uo "$IMAGES_REFS_FILE" "$IMAGES_REFS_FILE" # Remove duplicate entries
+  fi
+}
+
+function find_checksums_file() {
+  for file in "$@"; do
+    if [[ "${file}" == *"checksums.txt" ]]; then
+      echo "${file}"
+      return 0
+    fi
   done
-  sort -uo imagerefs.txt imagerefs.txt # Remove duplicate entries
+  warning "cannot find checksums file"
 }
 
 # Build a release from source.
 function sign_release() {
-  get_images_in_yamls "${ARTIFACTS_TO_PUBLISH}"
   if (( ! IS_PROW )); then # This function can't be run by devs on their laptops
     return 0
   fi
+  get_images_in_yamls "${ARTIFACTS_TO_PUBLISH}"
+  local checksums_file
+  checksums_file="$(find_checksums_file "${ARTIFACTS_TO_PUBLISH}")"
+
+  if ! [[ -f "${checksums_file}" ]]; then
+    echo '>> No checksums file found, generating one'
+    checksums_file="$(mktemp -d)/checksums.txt"
+    for file in ${ARTIFACTS_TO_PUBLISH}; do
+      sha256sum "${file}" >> "${checksums_file}"
```

Comment:
```suggestion
      pushd "$(dirname "$file")" >/dev/null
      sha256sum "$(basename "$file")" >> "${checksums}"
      popd >/dev/null
```

Reply:
0f24a18 fixes this

Reply:
Does this run into an issue if we have two files with the same name in subdirectories?  I think I'd rather `cd`/`pwd` into the root release directory and then use relative paths from there.

Also wondering how you feel about the `(cd ...; ...)` construct vs `pwd ...; ...; popd`

Reply:
The checksum should not have paths because we host it on GH release page, and it should be valid in that regard. It doesn't make sense to preserve full paths, as that makes validation very problematic for end users.

You can see such validation here: https://github.com/knative-sandbox/homebrew-kn-plugins/issues/134#issuecomment-1419171040

The `(cd ...; ...)` work similar to `pushd; ...; ...; popd`, although opening a subshell is tricky as you could lose changes done withing subshell (people don't expect sub-shells)